pkgs/top-level/python-packages.nix

```diff
@@ -5092,6 +5092,8 @@ self: super: with self; {
 
   htmllaundry = callPackage ../development/python-modules/htmllaundry { };
 
+  htmllistparse = callPackage ../development/python-modules/htmllistparse { };
+
```

Comment:
```suggestion
```

The package should be placed under the `pkgs/by-name/so/some-example/package.nix`hierarchy as it is the new preferred way for adding packages.

Reply:
Does this already apply for python packages, since they are evaluated in a slightly different way?

Reply:
If it is a library I'm actually unsure what the correct way of adding it is. I wrote the review with the assumption that it was a separate python program 🐍

Reply:
RFC 140 states that a limitation of it is that non-toplevel packages cannot be put into the structure yet (https://github.com/NixOS/rfcs/blob/master/rfcs/0140-simple-package-paths.md#drawbacks). I am not aware of any consensus on where to put it yet.

Even if it was a standalone python application, it would still have to reside in the python-modules folder and be called from python-packages.nix; since it requires the pythonPackages callPackage to be used.

Reply:
Thanks for sharing (:

Reply:
> Even if it was a standalone python application, it would still have to reside in the python-modules folder and be called from python-packages.nix; since it requires the pythonPackages callPackage to be used.

The python applications should not be in python-modules. It can be called from top-level using python3Packages.buildPythonApplication.

You can see some examples by running the following command.
`rg buildPythonApplication pkgs/by-name/`

Reply:
Ah, yes, makes sense. I was thinking of a python application like e.g. `bundlewrap`, which is both a library and an application.